### ISSUE TITLE
Attribution metric choice

### ISSUE THREAD

**Issue by aereeeee** [hannamw/MIB-circuit-track#22]
Hi, I have a question about the choice of attribution metric.

In your implementation, the default attribution metric is the `logit_diff`, while the evaluation uses a normalized metric [faithfulness](https://github.com/hannamw/MIB-circuit-track/blob/b759df34433c9e31043ba9e02908ce0bf20e894f/MIB_circuit_track/evaluation.py#L55) by default. Is there a particular reason why EAP-style methods don’t use the faithfulness metric directly for the gradient computation?

In my experiments, using faithfulness as the attribution metric yields much better performance on the evaluation. I may be missing some design consideration behind using logit_diff, so I wanted to check whether this choice was intentional and if there are tradeoffs I should be aware of.

Thanks in advance!

**Comment by aaronmueller** [comment]
This sounds similar conceptually to what the UGS method does (see [Li & Janson, 2024](https://arxiv.org/abs/2409.09951v1))! This method learns a mask over the model components to remove target information while preserving the model's general performance. The limitation of this method relative to gradient attribution is that it is quite memory- and compute-intensive, as we need to optimize a new parameter for every model component.

For attribution, we used the logit difference mainly because we wanted to benchmark the performance of the more common methods in the field. Most commonly, causal mediation analysis–style methods like these involve pre-computing the components that have the greatest influence on the logits of the correct answer, regardless of their relationship with other components. Using faithfulness as the target metric would allow one to consider multi-component relationships, but it also means that the method would no longer run in constant time, as we now have the sequential dependency of some circuit from a prior iteration C_t.

Your method sounds interesting; does it run in a similar time and using a similar amount of memory compared to the logic-based attribution?
